python_modules/dagster/dagster/_serdes/serdes.py

```diff
@@ -998,7 +997,9 @@ def _unpack_object(val: dict, whitelist_map: WhitelistMap, context: UnpackContex
 
     if "__mapping_items__" in val:
         return {
-            _unpack_value(k, whitelist_map, context): _unpack_value(v, whitelist_map, context)
+            cast(str, _unpack_value(k, whitelist_map, context)): _unpack_value(
+                v, whitelist_map, context
+            )
             for k, v in val["__mapping_items__"]
```

Comment:
I think this `str` cast is wrong, the key type type for this is non-scalar but still keyable https://github.com/dagster-io/dagster/pull/18057

Reply:
I see, OK I've changed to cast to `Any`.